app/controllers/provider_foreman_controller.rb

```diff
@@ -196,15 +183,9 @@ def cancel_provider_foreman
 
   def provider_foreman_form_fields
     assert_privileges("provider_foreman_edit_provider")
-    config_mgr = find_record(ManageIQ::Providers::ConfigurationManager, params[:id])
-    if config_mgr[:type] == "ManageIQ::Providers::AnsibleTower::ConfigurationManager"
-      provider = ManageIQ::Providers::AnsibleTower::Provider.find(config_mgr.provider_id)
-    else
-      provider = ManageIQ::Providers::Foreman::Provider.find(config_mgr.provider_id)
-    end
-    authentications = Authentication.where(:resource_id => provider[:id], :resource_type => "Provider")
+    authentications = find_record(ManageIQ::Providers::ConfigurationManager, params[:id]).provider.authentications
```

Comment:
this line should not be here ( I know it was there before your PR)
these lines should be here :
config_mgr = find_record(ManageIQ::Providers::ConfigurationManager, params[:id])
and also need the provider found for this config_mgr id.